raw-spans-grouper/raw-spans-grouper/src/main/java/org/hypertrace/core/rawspansgrouper/RawSpansGroupingTransformer.java

```diff
@@ -71,6 +83,17 @@ public void init(ProcessorContext context) {
   public KeyValue<String, StructuredTrace> transform(TraceIdentity key, RawSpan value) {
     ValueAndTimestamp<RawSpans> rawSpans = inflightTraceStore.get(key);
     RawSpans agg = rawSpans != null ? rawSpans.value() : RawSpans.newBuilder().build();
+    if (maxSpanCountMap.containsKey(key.getTenantId()) && agg.getRawSpans().size() > maxSpanCountMap.get(key.getTenantId())) {
+      if (logger.isDebugEnabled()) {
```

Comment:
as far as I remember if we use the `{}` wildcard we don't need to add this check cc @aaron-steinfeld

Reply:
Kind of - it depends what the reason for deferring is. If it's the `toString()` call, slf4j will do that for you if necessary as long as we use bindings instead of direct interpolation. Here, I'm guessing the HexUtils.getHex call is non trivial, so we'd want to avoid running that too (and thus keep the conditional). An alternative would be a wrapper object that takes a lambda (since this version of slf4j doesn't), but out of scope here.

That would look like: 
```java
public class MessageSupplier {
    private final Supplier<?> supplier;

    public MessageSupplier(Supplier<?> supplier) {
        this.supplier = supplier;
    }

    @Override
    public String toString() {
        return supplier.get().toString();
    }
}
```

Reply:
without the check, it will still evaluate the arguments. Although, `HexUtils.getHex()` is lightweight, this check will avoid it altogether.